canonical-web-and-design/snapcraft.io
# First snap flow - choose snap name

Fixes #1786

Adds a form to choose snap name at the beginning of the package step of first snap flow

### QA

- ./run or demo
- go to first snap flow
- in the beginning of the Package snap there should be a 'Choose a snap name' form
- choose a snap name, try using invalid name to see if validation works
- 'Choose' button should be disabled until valid name is used
- Click 'Choose snap name' button, page should reload, first step should be marked done, second step should be open
- Make sure that name chosen by user is used in next steps (in some warning notifications and commands)
- Chosen snap name should be used in the register name form on the last step of first snap flow
- Go back and choose different language for first snap flow
- Snap name chosen in different language should not be used

#### Check different language/OS configurations to see all is fine

<img width="1023" alt="Screenshot 2019-04-24 at 14 51 22" src="https://user-images.githubusercontent.com/83575/56661286-b8572f00-66a1-11e9-95b9-3910aa53cf7c.png">
